### PR TITLE
ddt removal in rhci tests

### PR DESCRIPTION
**Pull request by sthirugn**
I do not have test setup for rhci.  @anarang can you please review this PR and test these changes in your local setup? Thanks!

### CODE REVIEW

**Comment by sthirugn** [comment]
I am not sure if @anarang is around today, lets review/merge this PR and wait for jenkins results.

**Comment by elyezer** [comment]
ACK

File: tests/foreman/rhci/test_rhci.py [hunk 1]
```diff
@@ -76,26 +78,25 @@ def test_positive_update_name(self, data):
...
-            name=data['name'],
-            organization=self.org,
-            lifecycle_environment=self.lc_env,
-            deploy_rhev=True,
-            rhev_engine_host=self.rhev_engine_host,
-            rhev_storage_type='NFS',
-            rhev_engine_admin_password='changeme',
-        ).create()
-        self.assertEquals(deployment.name, data['name'])
+        for data in valid_name_update_tests():
```
Comment (oshtaier) [inline tests/foreman/rhci/test_rhci.py]: Suggesting removing that `valid_name_update_tests` and use same construction here as above:
```
    for new_name in (gen_string(str_type='alpha'),
                     gen_string(str_type='alphanumeric'),
                     gen_string(str_type='cjk'),
                     gen_string(str_type='latin1')):
```
We have already verified that entity can be created using any data type, so we can use `name=gen_string('alphanumeric') instead name=data['name']

**Comment by oshtaier** [comment]
ACK pending comment and reply from @anarang 
Please, try to run tests before merging

File: tests/foreman/rhci/test_rhci.py (hunk 1 above)
Comment (sthirugn) [inline tests/foreman/rhci/test_rhci.py]: I am not the right person to change the test criteria for `rhci` tests.  My only idea of this PR is to remove @ddt, @data usage so I can eventually remove ddt dependency for robottelo.

If this optimization is required, I will let @anarang do that.

**Comment by sthirugn** [comment]
As I said I have no setup for running rhci test and my intention is to remove ddt from this module.  I am okay with blocking this PR until @anarang finishes testing.

**Comment by oshtaier** [comment]
Sure. Let's block till that both things addressed by @anarang

**Comment by elyezer** [comment]
@sthirugn actually RHCI tests was developed by @JacobCallahan and @anarang takes care of the RHAI tests.

**Comment by sthirugn** [comment]
@elyezer you are right, sorry for the confusion

**Comment by sthirugn** [comment]
@JacobCallahan do you have setup/time to test this change?

**Comment by JacobCallahan** [comment]
@sthirugn I no longer have RHCI resources allocated to me.

**Comment by oshtaier** [comment]
@JacobCallahan Any ideas, what we can do to test that functionality?

**Comment by JacobCallahan** [comment]
@oshtaier Aside from contacting dajo to point the tests at a box, no. They mostly clean up after themselves so that is a bonus!

**Comment by oshtaier** [comment]
@sthirugn Is there a chance to reach that person? Do you like to do that?

**Comment by sthirugn** [comment]
Nope we are not running this on our composes as these tests are residing in tests/foreman/rhci/ package.  I am not sure who runs these tests and when.  With my current tasks, I cannot afford to spend time on following up / setting up rhci tests.  If any of you have bandwidth to do this testing, please feel free to do the same and if need be, this PR can stay open until then.

**Comment by oshtaier** [comment]
I do not propose to you do testing, but we need to figure out who is working on that

@omaciel Could you please, let us know who is working on RHCI package? And who have corresponding infrastructure? (even better :))

**Comment by sthirugn** [comment]
@omaciel this is waiting on you now :)

**Comment by omaciel** [comment]
@JacobCallahan can you please help them out with this?

**Comment by JacobCallahan** [comment]
Sure can!

File: tests/foreman/rhci/test_rhci.py [hunk 2]
```diff
@@ -76,26 +78,25 @@ def test_positive_update_name(self, data):
...
-        # perform the update
-        deployment.name = data['new-name']
-        deployment = deployment.update(['name'])
-        self.assertEqual(deployment.name, data['new-name'])
-        self.to_delete.append(deployment)
-
-    @data(
-        gen_string(str_type='alpha'),
-    )
     def test_positive_delete_deployment(self, data):
```
Comment (JacobCallahan) [inline tests/foreman/rhci/test_rhci.py]: remove data from the argument here and that will fix.
```TypeError: test_positive_delete_deployment() takes exactly 2 arguments (1 given)```

**Comment by JacobCallahan** [comment]
After that one change, you are good to go!
```
$ make test-foreman-rhci
python -m cProfile -o test-foreman-rhci.pstats $(which nosetests) --logging-filter=nailgun,robottelo --with-xunit --xunit-file=foreman-results.xml tests/foreman/rhci
....
----------------------------------------------------------------------
Ran 4 tests in 7.872s

OK
```

**Comment by sthirugn** [comment]
Thank you so much @JacobCallahan.  Review comments addressed and rebased.

**Comment by JacobCallahan** [comment]
ACK

**Comment by oshtaier** [comment]
@JacobCallahan Thank you for help

**Comment by sthirugn** [comment]
Part of https://github.com/SatelliteQE/robottelo/issues/2769

**Pull request merged**